airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -483,6 +484,7 @@ public KubePodProcess(final boolean isOrchestrator,
         .withNewMetadata()
         .withName(podName)
         .withLabels(labels)
+        .withAnnotations(annotations.orElse(null))
```

Comment:
what happens if this is an empty map or null? if the effect is the same, it'll be slightly cleaner to drop the optional and have `splitKVPairsFromEnvString`. Sorry, I know you are following an already existing pattern - however I think it complicates things. What do you think?

Reply:
Hi @davinchia, the optional has been added in this PR: https://github.com/airbytehq/airbyte/pull/10200 , but I also don't really see its benefit.
`splitKVPairsFromEnvString` (previously `getNodeSelectorsFromEnvString`) is now used for both the annotations and the node selectors.
So, you would suggest that `splitKVPairsFromEnvString` returns a simple `Map<String, String>` instead of `Optional<Map<String, String>>` and adapt its usage for the annotations and node selectors, correct?

Reply:
That was actually discussed in this comment: https://github.com/airbytehq/airbyte/pull/10200#discussion_r803239145

Reply:
yeap! returning a map would simplify the entire code execution chain.